src/main/java/com/cloudbees/jenkins/GitHubRepositoryNameContributor.java

```diff
@@ -30,13 +34,13 @@
      * Looks at the definition of {@link AbstractProject} and list up the related github repositories,
      * then puts them into the collection.
      */
-    public abstract void parseAssociatedNames(AbstractProject<?,?> job, Collection<GitHubRepositoryName> result);
+    public abstract void parseAssociatedNames(Job<?,?> job, Collection<GitHubRepositoryName> result);
 
     public static ExtensionList<GitHubRepositoryNameContributor> all() {
         return Jenkins.getInstance().getExtensionList(GitHubRepositoryNameContributor.class);
     }
 
-    public static Collection<GitHubRepositoryName> parseAssociatedNames(AbstractProject<?, ?> job) {
+    public static Collection<GitHubRepositoryName> parseAssociatedNames(Job<?, ?> job) {
```

Comment:
Same here, though easier since you need only introduce the new method and deprecate the existing one.